problem_5_3.py

```diff
@@ -0,0 +1,59 @@
+import pint
+def sr_of_earth():
+    ureg = pint.UnitRegistry(system='mks')#set default system so that the unit of radius is m
+    G = 6.67 * 10**(-11) * ureg.N * ureg.meter**2 / ureg.kg**2#gravitaitonal constant
+    m = 2 * 10**30 * ureg.kg#solar mass
+    c = 3 * 10**8 * ureg.meter / ureg.s#speed of light
+    r = 2 * G * m / c**2#Schwarzschild radius
+    print(r.to_base_units())
+sr_of_earth()
+
+import halo_mass_function as hmf
+import numpy as np
+def halonumber(n):
+	halo = hmf.HaloMassFunction(redshift=0)#set redshift = 0
+	h = 0.67#hubble constant
+	mass = np.array([12 + i * 8 / n for i in range(0, n+1)]) / h#upper limit of halo mass is 20 Msolar/h
+	dn = halo.dndm(mass)
+	number = (sum(2 * dn) - dn[0] -dn[-1]) * 8 / (2 * n * h)#use sum to calculate integral.
```

Comment:
This is giving the wrong answer! You have integrated from 12 - 20 Msun rather than 10^12 - 10^20 Msun. I think this might have been an attempt to use log-spacing (which is the right thing to do) that collided with a linear integration. Also, consider using a standard integration routine rather than rolling your own (although your hand-rolled routine is I think correct).